### PR TITLE
GitHub App configuration support

### PR DESCRIPTION
**Pull request by m1racoli**
This PR is the continuation of #46, after development has been stale for over two weeks.

Most of the work has been done by @blacksd and already reviewed by @AndreZiviani. I have only fixed style errors of the README.

Closes #46

### CODE REVIEW

**Comment by AndreZiviani** [comment]
Thanks for picking this up!

@lkysow can you approve the CI and review the PR please?

**Comment by m1racoli** [comment]
This still needs a review, right?

**Comment by mzupan** [comment]
only big issue is being able to pre-set a secret for the app pem file and secret key. So those are in clear text if you do a `helm get values atlantis`

File: charts/atlantis/templates/statefulset.yaml [hunk 1]
```diff
@@ -69,6 +69,16 @@ spec:
         secret:
           secretName: {{ template "atlantis.awsSecretName" . }}
       {{- end }}
+      {{- if .Values.githubApp }}
+      {{- if index .Values.githubApp "key.pem"}}
+      - name: github-app-key-volume
+        secret:
+          secretName: {{ template "atlantis.vcsSecretName" . }}
```
Comment (mzupan) [inline charts/atlantis/templates/statefulset.yaml:76]: needs `valueFrom`

File: charts/atlantis/templates/statefulset.yaml [hunk 2]
```diff
@@ -213,6 +223,24 @@ spec:
...
+          {{- if .Values.githubApp }}
+          {{- if .Values.githubApp.id }}
+          - name: ATLANTIS_GH_APP_ID
+            value: {{ .Values.githubApp.id | quote}}
+          {{- end }}
+          - name: ATLANTIS_WRITE_GIT_CREDS
+            value: "true"
+          {{- if .Values.githubApp.github_secret }}
+          - name: ATLANTIS_GH_WEBHOOK_SECRET
+            secretKeyRef:
```
Comment (mzupan) [inline charts/atlantis/templates/statefulset.yaml:235]: needs `valueFrom`

File: charts/atlantis/templates/statefulset.yaml (hunk 1 above)
Comment (m1racoli) [reply charts/atlantis/templates/statefulset.yaml:76]: This is not true for volumes. `valueFrom` is only needed for env vars.

**Comment by m1racoli** [comment]
> only big issue is being able to pre-set a secret for the app pem file and secret key. So those are in clear text if you do a `helm get values atlantis`

This is a general issue of helm. That's why helm state is encoded in secrets. Users without access to those secrets are not able to retrieve the deployed apps of the namespace.

Furthermore, this PR does not introduce this issue as other secrets (`aws_secret_access_key`,...) are already configurable.

**Comment by m1racoli** [comment]
@mzupan @lkysow what's blocking us from adding Github app support to the chart?

**Pull request merged**